keras/callbacks.py

```diff
@@ -2004,7 +2004,14 @@ def on_train_end(self, logs=None):
             io_utils.print_msg(
                 f"Epoch {self.stopped_epoch + 1}: early stopping"
             )
-
+		if self.restore_best_weights and self.best_weights is not None:
```

Comment:
Thanks for the PR! This is making sense to me. A couple of quick questions:
1) can you fix the indentation?
2) can you make this block of code a method on the class that can be shared here and `on_epoch_end`?
3) can you add a test to cover this piece of logic?

Reply:
1. yes I should be able to do it
2. I'm not very expert in coding so I would avoid making too many changes
3. I'm afraid not